crates/sui-framework/sources/dynamic_field.move

```diff
@@ -100,6 +125,18 @@ public fun remove<Name: copy + drop + store, Value: store>(
     value
 }
 
+spec remove {
+    pragma opaque;
+    aborts_if [abstract] !sui::prover::uid_has_field(object.id.bytes, name);
+    modifies [abstract] global<object::DynamicFields<Name>>(object.id.bytes);
+    ensures [abstract] object.id == old(object.id);
+    ensures [abstract] exists<object::DynamicFields<Name>>(object.id.bytes);
+    ensures [abstract] sui::prover::vec_remove(global<object::DynamicFields<Name>>(object.id.bytes).names,
+        index_of(global<object::DynamicFields<Name>>(object.id.bytes).names, name), 0) ==
+          old(global<object::DynamicFields<Name>>(object.id.bytes).names);
```

Comment:
Hmmm IIUC this is saying "removing the name from the new vec is equal to the old vec" but I would expect this one to say "removing the name from the old vec is equal to the new vec" instead. What am I missing?

Reply:
It is certainly I who missed it... I have no clue why the tests were passing (and not due to inconsistency I think as changing the expected length, which is expected to be one element shorter in the spec was causing a verification error). In any case, I am trying to fix it but clearly missing something as the `ensures !sui::prover::has_field(...)` clauses in the spec no longer hold (though the length check does...). Perhaps @meng-xu-cs can help debug it...